polkadot/node/network/collator-protocol/src/validator_side/tests/prospective_parachains.rs

```diff
@@ -1314,3 +1385,596 @@ fn child_blocked_from_seconding_by_parent(#[case] valid_parent: bool) {
 		virtual_overseer
 	});
 }
+
+#[test]
```

Comment:
Can you elaborate? 

If `advertisement_spam_protection` is extended to check the claim queue limit is respected it will overlap with `collations_outside_limits_are_not_fetched`. Imo they test different cases and it's nice to have them separated. The test names could be better though.